sourcegraph/sourcegraph
# Fuzzy finder: add personalized ranking based on history

Previously, the fuzzy finder did a bad job of surfacing frequently visited repos, symbols, and files. This PR adds personalized ranking by boosting results that the user has previously visited. This feature requires new infrastructure to collect the browsing history of the user. This PR stores the browser history in local storage but we may want to move this to a remote storage like temporary settings in the future.

Demo https://www.loom.com/share/54e39f710982443fa76689d5dd20153b


## Test plan

Start local server
```
SOURCEGRAPHDOTCOM_MODE=true SOURCEGRAPH_API_URL=https://sourcegraph.com sg start web-standalone
```

* Visit random repos and files
* Open fuzzy finder and observe that the visited repos and files get ranked at the top




## App preview:

- [Web](https://sg-web-olafurpg-fuzzy-history.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
